fcm/fcm.go

```diff
@@ -90,31 +90,44 @@ func (f *FCM) ConnectToFcm(fcmNotifications chan<- notification.PrinterNotificat
 	resp, err := http.Get(fmt.Sprintf("%s?token=%s", f.fcmServerBindURL, iidToken))
 	if err != nil {
 		// failed for ever no need to retry
+		log.Errorf("%v", err)
 		quit <- struct{}{}
 		return err
 	}
 	if resp.StatusCode == 200 {
 		reader := bufio.NewReader(resp.Body)
 		go func() {
 			for {
-				sizeB, err := reader.ReadBytes('\n')
-				if err == io.EOF {
-					log.Info("DRAIN message received, client reconnecting.")
-					dead <- struct{}{}
-					break
+				raw_input, err1 := reader.ReadBytes('\n')
```

Comment:
should you break out sooner on err1? For example, if err1 is EOF we know we have a problem (need to reconnect) and can skip the below Read call

Reply:
Fixed.